service/src/main/java/com/fecklessweasel/service/FileUploadServlet.java

```diff
@@ -0,0 +1,85 @@
+package com.fecklessweasel.service;
+
+import com.fecklessweasel.service.datatier.SQLInteractionInterface;
+import com.fecklessweasel.service.datatier.SQLSource;
+import com.fecklessweasel.service.objectmodel.FileMetadata;
+import com.fecklessweasel.service.objectmodel.ServiceException;
+import com.fecklessweasel.service.objectmodel.UserSession;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Date;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+@WebServlet("/servlet/file/upload")
+@MultipartConfig
+public class FileUploadServlet extends HttpServlet {
+
+    private static final String FILEPATH_PREFIX = "/files";
+
+    @Override
+    protected void doPost(final HttpServletRequest request,
+                          final HttpServletResponse response)
+            throws ServletException, IOException {
+
+        // TODO: These should go into database eventually
+        String title = request.getParameter("title");
+        String description = request.getParameter("description");
+
+        Part filePart = request.getPart("file[0]");
+        String fileName = filePart.getSubmittedFileName();
+
+        final UserSession session = UserSessionUtil.resumeSession(request);
+        // If user is not authenticated
+        if (session == null) {
+            return;
```

Comment:
Since this is a servlet it's probably save to assume that this is frontend code. Since they aren't authenticated perhaps we should redirect them to a page. Alternatively you can just throw a ServiceException() for NOT_AUTHENTICATED or something and the site will automatically redirect them.